src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.example.bookshop.service.impl;
+
+import com.example.bookshop.dto.BookDto;
+import com.example.bookshop.dto.CreateBookRequestDto;
+import com.example.bookshop.exception.EntityNotFoundException;
+import com.example.bookshop.mapper.BookMapper;
+import com.example.bookshop.model.Book;
+import com.example.bookshop.repository.BookRepository;
+import com.example.bookshop.service.BookService;
+import java.util.List;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toBook(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> getAll() {
+        return bookRepository.getAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<BookDto> findAllByAuthor(String author) {
+        return bookRepository.findAllByAuthor(author).stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(bookRepository
+                .findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book"
+                        + " with this id: " + id)));

```

Comment:
you can leverage Optional's `map` method:
```
return repo.findById(id)
        .map(mapper:toDto)
        .orElseThrow(...);
```

Reply:
About property: I guess, I saw it in one of the videos. I can remove it if you want it